contracts/staking/LPStaker.sol

```diff
@@ -0,0 +1,157 @@
+pragma solidity =0.6.12;
```

Comment:
I completely agree with all your concerns. But the purpose of developing this contract is to use the existing deployed staking contracts.
The contracts which you have written are good and will definitely solve some of the above mentioned concerns. But the thing is we will have redeploy all the staking contracts and ask users to unstake from existing contracts and stake in the new one. I think this is going to be a uphill task because 100s of millions are staked in our contracts right now. We may loose some users during this activity.

Another reason behind using this approach is to use the existing staking contracts because those are battle tested. As soon as we decide to change the staking contracts for any reasons we will use contract.

@mudgen

Reply:
@sameepsi Yes, makes total sense to me.

Reply:
ok